{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "62838af0-612f-427e-9247-b4fc24802f31",
   "metadata": {},
   "source": [
    "Create figures for scoping review"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "99bf7d6f-6a47-47e8-9588-0e7dfc12cbef",
   "metadata": {},
   "source": [
    "Import packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "edc0a6fe-f93b-42a9-84c0-cfd934c28f93",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Import packages\n",
    "library(data.table)\n",
    "library(ggplot2)\n",
    "library(dplyr)\n",
    "library(tidyr)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "88c5965c-fcf1-4d90-b91c-ff3e2830c56f",
   "metadata": {},
   "source": [
    "Import all the data:\n",
    "\n",
    "### META DATA\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "50fbc0af-f3f5-44ce-92cc-866adb946ab0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>studyRef</th><th scope=col>assessedBy</th><th scope=col>firstAuthor</th><th scope=col>year</th><th scope=col>species</th><th scope=col>countryStudy</th><th scope=col>mainOutcome</th><th scope=col>studyScore</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1182475512</td><td>AS/SE</td><td>O'Higgins  </td><td>2014</td><td>Human</td><td>Ireland </td><td>Both  </td><td>II </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1182475524</td><td>AS/SE</td><td>Cox        </td><td>2019</td><td>Human</td><td>Belgium </td><td>Both  </td><td>III</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1182476286</td><td>AS/SE</td><td>Assaf-Balut</td><td>2019</td><td>Human</td><td>Spain   </td><td>Both  </td><td>II </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1182476955</td><td>AS/SE</td><td>Assaf-Balut</td><td>2019</td><td>Human</td><td>Spain   </td><td>Both  </td><td>II </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1182477024</td><td>AS/SE</td><td>Fondjo     </td><td>2020</td><td>Human</td><td>Ghana   </td><td>Mother</td><td>II </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1182477025</td><td>AS/SE</td><td>Shamsi     </td><td>2010</td><td>Human</td><td>Pakistan</td><td>Mother</td><td>III</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & studyRef & assessedBy & firstAuthor & year & species & countryStudy & mainOutcome & studyScore\\\\\n",
       "  & <int> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1182475512 & AS/SE & O'Higgins   & 2014 & Human & Ireland  & Both   & II \\\\\n",
       "\t2 & 1182475524 & AS/SE & Cox         & 2019 & Human & Belgium  & Both   & III\\\\\n",
       "\t3 & 1182476286 & AS/SE & Assaf-Balut & 2019 & Human & Spain    & Both   & II \\\\\n",
       "\t4 & 1182476955 & AS/SE & Assaf-Balut & 2019 & Human & Spain    & Both   & II \\\\\n",
       "\t5 & 1182477024 & AS/SE & Fondjo      & 2020 & Human & Ghana    & Mother & II \\\\\n",
       "\t6 & 1182477025 & AS/SE & Shamsi      & 2010 & Human & Pakistan & Mother & III\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | studyRef &lt;int&gt; | assessedBy &lt;chr&gt; | firstAuthor &lt;chr&gt; | year &lt;int&gt; | species &lt;chr&gt; | countryStudy &lt;chr&gt; | mainOutcome &lt;chr&gt; | studyScore &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1182475512 | AS/SE | O'Higgins   | 2014 | Human | Ireland  | Both   | II  |\n",
       "| 2 | 1182475524 | AS/SE | Cox         | 2019 | Human | Belgium  | Both   | III |\n",
       "| 3 | 1182476286 | AS/SE | Assaf-Balut | 2019 | Human | Spain    | Both   | II  |\n",
       "| 4 | 1182476955 | AS/SE | Assaf-Balut | 2019 | Human | Spain    | Both   | II  |\n",
       "| 5 | 1182477024 | AS/SE | Fondjo      | 2020 | Human | Ghana    | Mother | II  |\n",
       "| 6 | 1182477025 | AS/SE | Shamsi      | 2010 | Human | Pakistan | Mother | III |\n",
       "\n"
      ],
      "text/plain": [
       "  studyRef   assessedBy firstAuthor year species countryStudy mainOutcome\n",
       "1 1182475512 AS/SE      O'Higgins   2014 Human   Ireland      Both       \n",
       "2 1182475524 AS/SE      Cox         2019 Human   Belgium      Both       \n",
       "3 1182476286 AS/SE      Assaf-Balut 2019 Human   Spain        Both       \n",
       "4 1182476955 AS/SE      Assaf-Balut 2019 Human   Spain        Both       \n",
       "5 1182477024 AS/SE      Fondjo      2020 Human   Ghana        Mother     \n",
       "6 1182477025 AS/SE      Shamsi      2010 Human   Pakistan     Mother     \n",
       "  studyScore\n",
       "1 II        \n",
       "2 III       \n",
       "3 II        \n",
       "4 II        \n",
       "5 II        \n",
       "6 III       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Meta data\n",
    "datMeta <- read.csv(\"tblStudyMeta.csv\", header=T)\n",
    "head(datMeta)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c96bf71-2def-42cb-b2a8-afbca1384804",
   "metadata": {},
   "source": [
    "\n",
    "### INFECTION DATA\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6f478816-9f39-453a-8a79-de33d99cfdff",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>studyRef</th><th scope=col>infectionType</th><th scope=col>X</th><th scope=col>X.1</th><th scope=col>X.2</th><th scope=col>X.3</th><th scope=col>X.4</th><th scope=col>X.5</th><th scope=col>X.6</th><th scope=col>X.7</th><th scope=col>X.8</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1182475512</td><td>Bacterial infection            </td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1182475524</td><td>Inflammatory/immune reponse    </td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1182476286</td><td>Urinary/Genital Tract Infection</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1182476955</td><td>Urinary/Genital Tract Infection</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1182477024</td><td>Malaria                        </td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1182477025</td><td>Urinary/Genital Tract Infection</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & studyRef & infectionType & X & X.1 & X.2 & X.3 & X.4 & X.5 & X.6 & X.7 & X.8\\\\\n",
       "  & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1182475512 & Bacterial infection             &  &  &  &  &  &  &  &  & \\\\\n",
       "\t2 & 1182475524 & Inflammatory/immune reponse     &  &  &  &  &  &  &  &  & \\\\\n",
       "\t3 & 1182476286 & Urinary/Genital Tract Infection &  &  &  &  &  &  &  &  & \\\\\n",
       "\t4 & 1182476955 & Urinary/Genital Tract Infection &  &  &  &  &  &  &  &  & \\\\\n",
       "\t5 & 1182477024 & Malaria                         &  &  &  &  &  &  &  &  & \\\\\n",
       "\t6 & 1182477025 & Urinary/Genital Tract Infection &  &  &  &  &  &  &  &  & \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | studyRef &lt;int&gt; | infectionType &lt;chr&gt; | X &lt;chr&gt; | X.1 &lt;chr&gt; | X.2 &lt;chr&gt; | X.3 &lt;chr&gt; | X.4 &lt;chr&gt; | X.5 &lt;chr&gt; | X.6 &lt;chr&gt; | X.7 &lt;chr&gt; | X.8 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1182475512 | Bacterial infection             | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> |\n",
       "| 2 | 1182475524 | Inflammatory/immune reponse     | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> |\n",
       "| 3 | 1182476286 | Urinary/Genital Tract Infection | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> |\n",
       "| 4 | 1182476955 | Urinary/Genital Tract Infection | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> |\n",
       "| 5 | 1182477024 | Malaria                         | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> |\n",
       "| 6 | 1182477025 | Urinary/Genital Tract Infection | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> | <!----> |\n",
       "\n"
      ],
      "text/plain": [
       "  studyRef   infectionType                   X X.1 X.2 X.3 X.4 X.5 X.6 X.7 X.8\n",
       "1 1182475512 Bacterial infection                                              \n",
       "2 1182475524 Inflammatory/immune reponse                                      \n",
       "3 1182476286 Urinary/Genital Tract Infection                                  \n",
       "4 1182476955 Urinary/Genital Tract Infection                                  \n",
       "5 1182477024 Malaria                                                          \n",
       "6 1182477025 Urinary/Genital Tract Infection                                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Infection data\n",
    "datInf<-read.csv(\"tblStudyInfections.csv\",header=T)\n",
    "head(datInf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b9100075-49d2-4f32-bcab-608b3209a1bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# long format with multiple infection types per study shown in multiple rows\n",
    "datInf_long <- datInf %>%\n",
    "  pivot_longer(cols = -studyRef, values_to = \"infectionType\") %>%\n",
    "  mutate(infectionType = na_if(infectionType, \"\")) %>%\n",
    "  filter(!is.na(infectionType))\n",
    "\n",
    "#trim white space\n",
    "datInf_long$infectionType <-trimws(datInf_long$infectionType)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e952cac8-b8fe-4881-8b7d-87ab90cf6907",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Bacterial infection'</li><li>'Inflammatory/immune reponse'</li><li>'Urinary/Genital Tract Infection'</li><li>'Malaria'</li><li>'Respiratory infection'</li><li>'Fever or cold'</li><li>'Other'</li><li>'Influenza'</li><li>'Periodontal'</li><li>'Covid-19'</li><li>'LPS injection'</li><li>'HIV'</li><li>'Parasitic infection (nematode/hookworm etc)'</li><li>'H. pylori infection'</li><li>'Hepatitis B/C'</li><li>'STI'</li><li>'Vaccine response'</li><li>'C-reactive protein'</li><li>'Intra-amniotic/intra-uterine infection (chorioamnionitis)'</li><li>'Group B streptococcus'</li><li>'Toxoplasmosis'</li><li>'Herpes virus (HSV)'</li><li>'Rubella immune status'</li><li>'Unspecified maternal infection'</li><li>'Pyelonephritis (Kidney infection)'</li><li>'Zika virus'</li><li>'Sepsis'</li><li>'Yeast infection'</li><li>'Autoimmune disorder'</li><li>'HPV'</li><li>'Systemic infection'</li><li>'Tuberculosis'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Bacterial infection'\n",
       "\\item 'Inflammatory/immune reponse'\n",
       "\\item 'Urinary/Genital Tract Infection'\n",
       "\\item 'Malaria'\n",
       "\\item 'Respiratory infection'\n",
       "\\item 'Fever or cold'\n",
       "\\item 'Other'\n",
       "\\item 'Influenza'\n",
       "\\item 'Periodontal'\n",
       "\\item 'Covid-19'\n",
       "\\item 'LPS injection'\n",
       "\\item 'HIV'\n",
       "\\item 'Parasitic infection (nematode/hookworm etc)'\n",
       "\\item 'H. pylori infection'\n",
       "\\item 'Hepatitis B/C'\n",
       "\\item 'STI'\n",
       "\\item 'Vaccine response'\n",
       "\\item 'C-reactive protein'\n",
       "\\item 'Intra-amniotic/intra-uterine infection (chorioamnionitis)'\n",
       "\\item 'Group B streptococcus'\n",
       "\\item 'Toxoplasmosis'\n",
       "\\item 'Herpes virus (HSV)'\n",
       "\\item 'Rubella immune status'\n",
       "\\item 'Unspecified maternal infection'\n",
       "\\item 'Pyelonephritis (Kidney infection)'\n",
       "\\item 'Zika virus'\n",
       "\\item 'Sepsis'\n",
       "\\item 'Yeast infection'\n",
       "\\item 'Autoimmune disorder'\n",
       "\\item 'HPV'\n",
       "\\item 'Systemic infection'\n",
       "\\item 'Tuberculosis'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Bacterial infection'\n",
       "2. 'Inflammatory/immune reponse'\n",
       "3. 'Urinary/Genital Tract Infection'\n",
       "4. 'Malaria'\n",
       "5. 'Respiratory infection'\n",
       "6. 'Fever or cold'\n",
       "7. 'Other'\n",
       "8. 'Influenza'\n",
       "9. 'Periodontal'\n",
       "10. 'Covid-19'\n",
       "11. 'LPS injection'\n",
       "12. 'HIV'\n",
       "13. 'Parasitic infection (nematode/hookworm etc)'\n",
       "14. 'H. pylori infection'\n",
       "15. 'Hepatitis B/C'\n",
       "16. 'STI'\n",
       "17. 'Vaccine response'\n",
       "18. 'C-reactive protein'\n",
       "19. 'Intra-amniotic/intra-uterine infection (chorioamnionitis)'\n",
       "20. 'Group B streptococcus'\n",
       "21. 'Toxoplasmosis'\n",
       "22. 'Herpes virus (HSV)'\n",
       "23. 'Rubella immune status'\n",
       "24. 'Unspecified maternal infection'\n",
       "25. 'Pyelonephritis (Kidney infection)'\n",
       "26. 'Zika virus'\n",
       "27. 'Sepsis'\n",
       "28. 'Yeast infection'\n",
       "29. 'Autoimmune disorder'\n",
       "30. 'HPV'\n",
       "31. 'Systemic infection'\n",
       "32. 'Tuberculosis'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Bacterial infection\"                                      \n",
       " [2] \"Inflammatory/immune reponse\"                              \n",
       " [3] \"Urinary/Genital Tract Infection\"                          \n",
       " [4] \"Malaria\"                                                  \n",
       " [5] \"Respiratory infection\"                                    \n",
       " [6] \"Fever or cold\"                                            \n",
       " [7] \"Other\"                                                    \n",
       " [8] \"Influenza\"                                                \n",
       " [9] \"Periodontal\"                                              \n",
       "[10] \"Covid-19\"                                                 \n",
       "[11] \"LPS injection\"                                            \n",
       "[12] \"HIV\"                                                      \n",
       "[13] \"Parasitic infection (nematode/hookworm etc)\"              \n",
       "[14] \"H. pylori infection\"                                      \n",
       "[15] \"Hepatitis B/C\"                                            \n",
       "[16] \"STI\"                                                      \n",
       "[17] \"Vaccine response\"                                         \n",
       "[18] \"C-reactive protein\"                                       \n",
       "[19] \"Intra-amniotic/intra-uterine infection (chorioamnionitis)\"\n",
       "[20] \"Group B streptococcus\"                                    \n",
       "[21] \"Toxoplasmosis\"                                            \n",
       "[22] \"Herpes virus (HSV)\"                                       \n",
       "[23] \"Rubella immune status\"                                    \n",
       "[24] \"Unspecified maternal infection\"                           \n",
       "[25] \"Pyelonephritis (Kidney infection)\"                        \n",
       "[26] \"Zika virus\"                                               \n",
       "[27] \"Sepsis\"                                                   \n",
       "[28] \"Yeast infection\"                                          \n",
       "[29] \"Autoimmune disorder\"                                      \n",
       "[30] \"HPV\"                                                      \n",
       "[31] \"Systemic infection\"                                       \n",
       "[32] \"Tuberculosis\"                                             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>infectionType</th><th scope=col>studyRef</th><th scope=col>InfCat</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>477</th><td>LPS injection                              </td><td>1182475095</td><td>Immune   </td></tr>\n",
       "\t<tr><th scope=row>275</th><td>Inflammatory/immune reponse                </td><td>1182475123</td><td>Immune   </td></tr>\n",
       "\t<tr><th scope=row>630</th><td>Parasitic infection (nematode/hookworm etc)</td><td>1182475134</td><td>Parasitic</td></tr>\n",
       "\t<tr><th scope=row>262</th><td>HIV                                        </td><td>1182475144</td><td>Viral    </td></tr>\n",
       "\t<tr><th scope=row>505</th><td>Malaria                                    </td><td>1182475144</td><td>Parasitic</td></tr>\n",
       "\t<tr><th scope=row>533</th><td>Malaria                                    </td><td>1182475147</td><td>Parasitic</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & infectionType & studyRef & InfCat\\\\\n",
       "  & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t477 & LPS injection                               & 1182475095 & Immune   \\\\\n",
       "\t275 & Inflammatory/immune reponse                 & 1182475123 & Immune   \\\\\n",
       "\t630 & Parasitic infection (nematode/hookworm etc) & 1182475134 & Parasitic\\\\\n",
       "\t262 & HIV                                         & 1182475144 & Viral    \\\\\n",
       "\t505 & Malaria                                     & 1182475144 & Parasitic\\\\\n",
       "\t533 & Malaria                                     & 1182475147 & Parasitic\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | infectionType &lt;chr&gt; | studyRef &lt;int&gt; | InfCat &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 477 | LPS injection                               | 1182475095 | Immune    |\n",
       "| 275 | Inflammatory/immune reponse                 | 1182475123 | Immune    |\n",
       "| 630 | Parasitic infection (nematode/hookworm etc) | 1182475134 | Parasitic |\n",
       "| 262 | HIV                                         | 1182475144 | Viral     |\n",
       "| 505 | Malaria                                     | 1182475144 | Parasitic |\n",
       "| 533 | Malaria                                     | 1182475147 | Parasitic |\n",
       "\n"
      ],
      "text/plain": [
       "    infectionType                               studyRef   InfCat   \n",
       "477 LPS injection                               1182475095 Immune   \n",
       "275 Inflammatory/immune reponse                 1182475123 Immune   \n",
       "630 Parasitic infection (nematode/hookworm etc) 1182475134 Parasitic\n",
       "262 HIV                                         1182475144 Viral    \n",
       "505 Malaria                                     1182475144 Parasitic\n",
       "533 Malaria                                     1182475147 Parasitic"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Categorise Infection data\n",
    "\n",
    "#What are all the Infection types?\n",
    "unique(datInf_long$infectionType)\n",
    "\n",
    "## Table catagorising infection types\n",
    "infection_lookup <- data.frame(\n",
    "  Infection = c(\"HPV\", \"Yeast infection\", \"Sepsis\", \"Systemic infection\", \"Tuberculosis\",\n",
    "                \"Toxoplasmosis\", \"Zika virus\", \"Autoimmune disorder\", \"Rubella immune status\",\n",
    "                \"Herpes virus (HSV)\", \"Influenza\", \"Pyelonephritis (Kidney infection)\", \"Fever or cold\",\n",
    "                \"Periodontal\", \"Group B streptococcus\", \"H. pylori infection\", \"Vaccine response\",\n",
    "                \"Bacterial infection\", \"LPS injection\", \"C-reactive protein\",\n",
    "                \"Unspecified maternal infection\", \"Hepatitis B/C\", \"Intra-amniotic/intra-uterine infection (chorioamnionitis)\",\n",
    "                \"STI\", \"Respiratory infection\", \"Other\", \"Malaria\", \n",
    "                \"Parasitic infection (nematode/hookworm etc)\", \"HIV\", \"Covid-19\", \"Urinary/Genital Tract Infection\",\n",
    "                \"Inflammatory/immune reponse\"),\n",
    "  InfCat = c(\"Viral\", \"Fungal\", \"Bacterial\", \"General\", \"Bacterial\",\n",
    "             \"Parasitic\", \"Viral\", \"Immune\", \"Viral\",\n",
    "             \"Viral\", \"Viral\", \"Bacterial\", \"General\",\n",
    "             \"General\", \"Bacterial\", \"Bacterial\", \"Immune\",\n",
    "             \"Bacterial\", \"Immune\", \"Immune\",\n",
    "             \"General\", \"Viral\", \"Bacterial\",\n",
    "             \"Bacterial\", \"Bacterial\", \"General\", \"Parasitic\",\n",
    "             \"Parasitic\", \"Viral\", \"Viral\", \"Bacterial\", \"Immune\")\n",
    ")\n",
    "\n",
    "#Check that all the infection types have been categorised\n",
    "setdiff(unique(datInf_long$infectionType), unique(infection_lookup$Infection))\n",
    "setdiff(unique(infection_lookup$Infection), unique(datInf_long$infectionType))\n",
    "\n",
    "#Create catagorised dataset\n",
    "# Left join infection_lookup with datInf\n",
    "datInfType_long <- merge(datInf_long, infection_lookup, by.x = \"infectionType\", by.y = \"Infection\", all.x = TRUE)\n",
    "datInfType_long <- datInfType_long[,-3]\n",
    "head(datInfType_long[order(datInfType_long$studyRef),])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "eb786079-ae2a-40ce-bb0d-cef66c785a7f",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "### NUTRITION DATA\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8684a195-dbf9-4d07-9917-859b15f4d836",
   "metadata": {},
   "outputs": [],
   "source": [
    "datNut<-read.csv(\"tblStudyNutrition.csv\",header=T)\n",
    "\n",
    "# long format with multiple nutrient types per study shown in multiple rows\n",
    "datNut_long <- datNut %>%\n",
    "  pivot_longer(cols = -studyRef, values_to = \"nutritionType\") %>%\n",
    "  mutate(nutritionType = na_if(nutritionType, \"\")) %>%\n",
    "  filter(!is.na(nutritionType))\n",
    "\n",
    "#trim white space\n",
    "datNut_long$nutritionType <-trimws(datNut_long$nutritionType)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f5b5a5ff-833f-4fc1-961b-3c48cdce5517",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'BMI'</li><li>'Dietary diversity/change'</li><li>'Anaemia'</li><li>'Folic acid'</li><li>'Iron'</li><li>'Mid-upper arm circumference (MUAC)'</li><li>'Vitamin A'</li><li>'Other'</li><li>'Vitamin B2/B6/B12'</li><li>'Zinc'</li><li>'Vitamin D'</li><li>'Vitamin C'</li><li>'Magnesium'</li><li>'Vitamin E'</li><li>'Fatty acids'</li><li>'Calcium'</li><li>'Copper'</li><li>'Boron'</li><li>'Protein'</li><li>'Multivitamin/micronutrient supplement etc'</li><li>'High fat diet'</li><li>'Gestational weight gain'</li><li>'Selenium'</li><li>'Fibre'</li><li>'Phosphorus'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'BMI'\n",
       "\\item 'Dietary diversity/change'\n",
       "\\item 'Anaemia'\n",
       "\\item 'Folic acid'\n",
       "\\item 'Iron'\n",
       "\\item 'Mid-upper arm circumference (MUAC)'\n",
       "\\item 'Vitamin A'\n",
       "\\item 'Other'\n",
       "\\item 'Vitamin B2/B6/B12'\n",
       "\\item 'Zinc'\n",
       "\\item 'Vitamin D'\n",
       "\\item 'Vitamin C'\n",
       "\\item 'Magnesium'\n",
       "\\item 'Vitamin E'\n",
       "\\item 'Fatty acids'\n",
       "\\item 'Calcium'\n",
       "\\item 'Copper'\n",
       "\\item 'Boron'\n",
       "\\item 'Protein'\n",
       "\\item 'Multivitamin/micronutrient supplement etc'\n",
       "\\item 'High fat diet'\n",
       "\\item 'Gestational weight gain'\n",
       "\\item 'Selenium'\n",
       "\\item 'Fibre'\n",
       "\\item 'Phosphorus'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'BMI'\n",
       "2. 'Dietary diversity/change'\n",
       "3. 'Anaemia'\n",
       "4. 'Folic acid'\n",
       "5. 'Iron'\n",
       "6. 'Mid-upper arm circumference (MUAC)'\n",
       "7. 'Vitamin A'\n",
       "8. 'Other'\n",
       "9. 'Vitamin B2/B6/B12'\n",
       "10. 'Zinc'\n",
       "11. 'Vitamin D'\n",
       "12. 'Vitamin C'\n",
       "13. 'Magnesium'\n",
       "14. 'Vitamin E'\n",
       "15. 'Fatty acids'\n",
       "16. 'Calcium'\n",
       "17. 'Copper'\n",
       "18. 'Boron'\n",
       "19. 'Protein'\n",
       "20. 'Multivitamin/micronutrient supplement etc'\n",
       "21. 'High fat diet'\n",
       "22. 'Gestational weight gain'\n",
       "23. 'Selenium'\n",
       "24. 'Fibre'\n",
       "25. 'Phosphorus'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"BMI\"                                      \n",
       " [2] \"Dietary diversity/change\"                 \n",
       " [3] \"Anaemia\"                                  \n",
       " [4] \"Folic acid\"                               \n",
       " [5] \"Iron\"                                     \n",
       " [6] \"Mid-upper arm circumference (MUAC)\"       \n",
       " [7] \"Vitamin A\"                                \n",
       " [8] \"Other\"                                    \n",
       " [9] \"Vitamin B2/B6/B12\"                        \n",
       "[10] \"Zinc\"                                     \n",
       "[11] \"Vitamin D\"                                \n",
       "[12] \"Vitamin C\"                                \n",
       "[13] \"Magnesium\"                                \n",
       "[14] \"Vitamin E\"                                \n",
       "[15] \"Fatty acids\"                              \n",
       "[16] \"Calcium\"                                  \n",
       "[17] \"Copper\"                                   \n",
       "[18] \"Boron\"                                    \n",
       "[19] \"Protein\"                                  \n",
       "[20] \"Multivitamin/micronutrient supplement etc\"\n",
       "[21] \"High fat diet\"                            \n",
       "[22] \"Gestational weight gain\"                  \n",
       "[23] \"Selenium\"                                 \n",
       "[24] \"Fibre\"                                    \n",
       "[25] \"Phosphorus\"                               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>nutritionType</th><th scope=col>studyRef</th><th scope=col>Category</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>683</th><td>Iron      </td><td>1182475095</td><td>Micronutrients </td></tr>\n",
       "\t<tr><th scope=row>82</th><td>BMI       </td><td>1182475123</td><td>Anthropometrics</td></tr>\n",
       "\t<tr><th scope=row>606</th><td>Folic acid</td><td>1182475134</td><td>Micronutrients </td></tr>\n",
       "\t<tr><th scope=row>680</th><td>Iron      </td><td>1182475134</td><td>Micronutrients </td></tr>\n",
       "\t<tr><th scope=row>810</th><td>Other     </td><td>1182475134</td><td>Diet/Other     </td></tr>\n",
       "\t<tr><th scope=row>898</th><td>Vitamin A </td><td>1182475134</td><td>Micronutrients </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & nutritionType & studyRef & Category\\\\\n",
       "  & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t683 & Iron       & 1182475095 & Micronutrients \\\\\n",
       "\t82 & BMI        & 1182475123 & Anthropometrics\\\\\n",
       "\t606 & Folic acid & 1182475134 & Micronutrients \\\\\n",
       "\t680 & Iron       & 1182475134 & Micronutrients \\\\\n",
       "\t810 & Other      & 1182475134 & Diet/Other     \\\\\n",
       "\t898 & Vitamin A  & 1182475134 & Micronutrients \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | nutritionType &lt;chr&gt; | studyRef &lt;int&gt; | Category &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 683 | Iron       | 1182475095 | Micronutrients  |\n",
       "| 82 | BMI        | 1182475123 | Anthropometrics |\n",
       "| 606 | Folic acid | 1182475134 | Micronutrients  |\n",
       "| 680 | Iron       | 1182475134 | Micronutrients  |\n",
       "| 810 | Other      | 1182475134 | Diet/Other      |\n",
       "| 898 | Vitamin A  | 1182475134 | Micronutrients  |\n",
       "\n"
      ],
      "text/plain": [
       "    nutritionType studyRef   Category       \n",
       "683 Iron          1182475095 Micronutrients \n",
       "82  BMI           1182475123 Anthropometrics\n",
       "606 Folic acid    1182475134 Micronutrients \n",
       "680 Iron          1182475134 Micronutrients \n",
       "810 Other         1182475134 Diet/Other     \n",
       "898 Vitamin A     1182475134 Micronutrients "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#What are all the nutritionTypes?\n",
    "unique(datNut_long$nutritionType)\n",
    "\n",
    "# add categorisation\n",
    "# Lookup table with shortened category names\n",
    "\n",
    "##Notes: \n",
    "#### Anaemia has been changed to a Clinical Indicator rather than Anthromometric\n",
    "#### Minerals and Vitamins have been combined into Micronutrients\n",
    "#### Because of the above combination we can now add Multivitamin/micronutrient supplement to Micronutrients. \n",
    "\n",
    "nutrient_lookup <- data.frame(\n",
    "  Nutrition = c(\"Boron\", \"Phosphorus\", \"Magnesium\", \"Calcium\", \"Copper\", \"Selenium\", \"Zinc\", \"Iron\",\n",
    "                \"Vitamin A\", \"Vitamin B2/B6/B12\", \"Vitamin C\", \"Vitamin D\", \"Vitamin E\", \"Folic acid\",\n",
    "                \"Protein\", \"Fatty acids\", \"Fibre\", \"High fat diet\",\n",
    "                \"Multivitamin/micronutrient supplement etc\",\n",
    "                \"BMI\", \"Gestational weight gain\", \"Mid-upper arm circumference (MUAC)\", \"Anaemia\",\n",
    "                \"Dietary diversity/change\", \"Other\"),\n",
    "  Category = c(\n",
    "    \"Micronutrients\", \"Micronutrients\", \"Micronutrients\", \"Micronutrients\",\n",
    "    \"Micronutrients\", \"Micronutrients\", \"Micronutrients\", \"Micronutrients\",\n",
    "    \n",
    "    \"Micronutrients\", \"Micronutrients\", \"Micronutrients\", \"Micronutrients\",\n",
    "    \"Micronutrients\", \"Micronutrients\",\n",
    "    \n",
    "    \"Macronutrients\", \"Macronutrients\", \"Macronutrients\", \"Macronutrients\",\n",
    "    \n",
    "    \"Micronutrients\",\n",
    "    \n",
    "    \"Anthropometrics\", \"Anthropometrics\", \"Anthropometrics\", \"Clinical Indicator\",\n",
    "    \n",
    "    \"Diet/Other\", \"Diet/Other\"\n",
    "  ),\n",
    "  stringsAsFactors = FALSE\n",
    ")\n",
    "\n",
    "\n",
    "\n",
    "#Check that all the nutrition types have been categorised\n",
    "setdiff(unique(datNut_long$nutritionType), unique(nutrient_lookup$Nutrition))\n",
    "setdiff(unique(nutrient_lookup$Nutrition), unique(datNut_long$nutritionType))\n",
    "\n",
    "#Create catagorised dataset\n",
    "# Left join nutrient_lookup with datNut\n",
    "datNutType_long <- merge(datNut_long, nutrient_lookup, by.x = \"nutritionType\", by.y = \"Nutrition\", all.x = TRUE)\n",
    "datNutType_long <- datNutType_long[,-3] #remove \"name\" column\n",
    "head(datNutType_long[order(datNutType_long$studyRef),])\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "16c95d49-7411-4991-9bfb-3d9cf1e207ed",
   "metadata": {},
   "source": [
    "\n",
    "### PREGNANCY OUTCOMES\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "79b3ed4f-7252-4f27-8cdf-a9fe069cfc17",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>studyRef</th><th scope=col>maternalOutcomeType</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1182475095</td><td>Biomarkers of immunity - maternal</td></tr>\n",
       "\t<tr><td>1182475123</td><td>Placental inflammation           </td></tr>\n",
       "\t<tr><td>1182475123</td><td>Other - maternal                 </td></tr>\n",
       "\t<tr><td>1182475134</td><td>Anaemia - maternal               </td></tr>\n",
       "\t<tr><td>1182475144</td><td>Anaemia - maternal               </td></tr>\n",
       "\t<tr><td>1182475147</td><td>Anaemia - maternal               </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " studyRef & maternalOutcomeType\\\\\n",
       " <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1182475095 & Biomarkers of immunity - maternal\\\\\n",
       "\t 1182475123 & Placental inflammation           \\\\\n",
       "\t 1182475123 & Other - maternal                 \\\\\n",
       "\t 1182475134 & Anaemia - maternal               \\\\\n",
       "\t 1182475144 & Anaemia - maternal               \\\\\n",
       "\t 1182475147 & Anaemia - maternal               \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| studyRef &lt;int&gt; | maternalOutcomeType &lt;chr&gt; |\n",
       "|---|---|\n",
       "| 1182475095 | Biomarkers of immunity - maternal |\n",
       "| 1182475123 | Placental inflammation            |\n",
       "| 1182475123 | Other - maternal                  |\n",
       "| 1182475134 | Anaemia - maternal                |\n",
       "| 1182475144 | Anaemia - maternal                |\n",
       "| 1182475147 | Anaemia - maternal                |\n",
       "\n"
      ],
      "text/plain": [
       "  studyRef   maternalOutcomeType              \n",
       "1 1182475095 Biomarkers of immunity - maternal\n",
       "2 1182475123 Placental inflammation           \n",
       "3 1182475123 Other - maternal                 \n",
       "4 1182475134 Anaemia - maternal               \n",
       "5 1182475144 Anaemia - maternal               \n",
       "6 1182475147 Anaemia - maternal               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>studyRef</th><th scope=col>offspringOutcomeType</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1182475095</td><td>Behavioural change              </td></tr>\n",
       "\t<tr><td>1182475159</td><td>Birth weight                    </td></tr>\n",
       "\t<tr><td>1182475165</td><td>Birth weight                    </td></tr>\n",
       "\t<tr><td>1182475165</td><td>Gestational age (Large or Small)</td></tr>\n",
       "\t<tr><td>1182475165</td><td>Neonate length                  </td></tr>\n",
       "\t<tr><td>1182475165</td><td>Head circumference              </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " studyRef & offspringOutcomeType\\\\\n",
       " <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1182475095 & Behavioural change              \\\\\n",
       "\t 1182475159 & Birth weight                    \\\\\n",
       "\t 1182475165 & Birth weight                    \\\\\n",
       "\t 1182475165 & Gestational age (Large or Small)\\\\\n",
       "\t 1182475165 & Neonate length                  \\\\\n",
       "\t 1182475165 & Head circumference              \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| studyRef &lt;int&gt; | offspringOutcomeType &lt;chr&gt; |\n",
       "|---|---|\n",
       "| 1182475095 | Behavioural change               |\n",
       "| 1182475159 | Birth weight                     |\n",
       "| 1182475165 | Birth weight                     |\n",
       "| 1182475165 | Gestational age (Large or Small) |\n",
       "| 1182475165 | Neonate length                   |\n",
       "| 1182475165 | Head circumference               |\n",
       "\n"
      ],
      "text/plain": [
       "  studyRef   offspringOutcomeType            \n",
       "1 1182475095 Behavioural change              \n",
       "2 1182475159 Birth weight                    \n",
       "3 1182475165 Birth weight                    \n",
       "4 1182475165 Gestational age (Large or Small)\n",
       "5 1182475165 Neonate length                  \n",
       "6 1182475165 Head circumference              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Birth weight'</li><li>'Apgar score'</li><li>'Gestational age (Large or Small)'</li><li>'NICU admission'</li><li>'Other - maternal'</li><li>'Gestational age (Days/Weeks)'</li><li>'Cord blood pH'</li><li>'Respiratory distress'</li><li>'Congenital malformations'</li><li>'Congenital heart disease'</li><li>'Growth'</li><li>'Macrosomia'</li><li>'Foetal distress'</li><li>'Neonatal death'</li><li>'COVID status'</li><li>'Behavioural change'</li><li>'Neonate length'</li><li>'Head circumference'</li><li>'Cleft palate'</li><li>'Microcephaly'</li><li>'Gene expression - offspring'</li><li>'Language and motor development'</li><li>'Biomarkers of immunity - offspring'</li><li>'Autism spectrum disorder'</li><li>'Motor development'</li><li>'Neonatal encephalopathy'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Birth weight'\n",
       "\\item 'Apgar score'\n",
       "\\item 'Gestational age (Large or Small)'\n",
       "\\item 'NICU admission'\n",
       "\\item 'Other - maternal'\n",
       "\\item 'Gestational age (Days/Weeks)'\n",
       "\\item 'Cord blood pH'\n",
       "\\item 'Respiratory distress'\n",
       "\\item 'Congenital malformations'\n",
       "\\item 'Congenital heart disease'\n",
       "\\item 'Growth'\n",
       "\\item 'Macrosomia'\n",
       "\\item 'Foetal distress'\n",
       "\\item 'Neonatal death'\n",
       "\\item 'COVID status'\n",
       "\\item 'Behavioural change'\n",
       "\\item 'Neonate length'\n",
       "\\item 'Head circumference'\n",
       "\\item 'Cleft palate'\n",
       "\\item 'Microcephaly'\n",
       "\\item 'Gene expression - offspring'\n",
       "\\item 'Language and motor development'\n",
       "\\item 'Biomarkers of immunity - offspring'\n",
       "\\item 'Autism spectrum disorder'\n",
       "\\item 'Motor development'\n",
       "\\item 'Neonatal encephalopathy'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Birth weight'\n",
       "2. 'Apgar score'\n",
       "3. 'Gestational age (Large or Small)'\n",
       "4. 'NICU admission'\n",
       "5. 'Other - maternal'\n",
       "6. 'Gestational age (Days/Weeks)'\n",
       "7. 'Cord blood pH'\n",
       "8. 'Respiratory distress'\n",
       "9. 'Congenital malformations'\n",
       "10. 'Congenital heart disease'\n",
       "11. 'Growth'\n",
       "12. 'Macrosomia'\n",
       "13. 'Foetal distress'\n",
       "14. 'Neonatal death'\n",
       "15. 'COVID status'\n",
       "16. 'Behavioural change'\n",
       "17. 'Neonate length'\n",
       "18. 'Head circumference'\n",
       "19. 'Cleft palate'\n",
       "20. 'Microcephaly'\n",
       "21. 'Gene expression - offspring'\n",
       "22. 'Language and motor development'\n",
       "23. 'Biomarkers of immunity - offspring'\n",
       "24. 'Autism spectrum disorder'\n",
       "25. 'Motor development'\n",
       "26. 'Neonatal encephalopathy'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Birth weight\"                       \"Apgar score\"                       \n",
       " [3] \"Gestational age (Large or Small)\"   \"NICU admission\"                    \n",
       " [5] \"Other - maternal\"                   \"Gestational age (Days/Weeks)\"      \n",
       " [7] \"Cord blood pH\"                      \"Respiratory distress\"              \n",
       " [9] \"Congenital malformations\"           \"Congenital heart disease\"          \n",
       "[11] \"Growth\"                             \"Macrosomia\"                        \n",
       "[13] \"Foetal distress\"                    \"Neonatal death\"                    \n",
       "[15] \"COVID status\"                       \"Behavioural change\"                \n",
       "[17] \"Neonate length\"                     \"Head circumference\"                \n",
       "[19] \"Cleft palate\"                       \"Microcephaly\"                      \n",
       "[21] \"Gene expression - offspring\"        \"Language and motor development\"    \n",
       "[23] \"Biomarkers of immunity - offspring\" \"Autism spectrum disorder\"          \n",
       "[25] \"Motor development\"                  \"Neonatal encephalopathy\"           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## This data was collected as MaternalOutcomes and OffspringOutcomes. We will combine these data into a single PregnancyOutcome column\n",
    "\n",
    "\n",
    "######## Maternal data\n",
    "datMat<-read.csv(\"tblStudyMaternalOutcomes.csv\",header=T)\n",
    "\n",
    "# long format with multiple nutrient types per study shown in multiple rows\n",
    "datMat_long <- datMat %>%\n",
    "  pivot_longer(cols = -studyRef, values_to = \"maternalOutcomeType\") %>%\n",
    "  mutate(maternalOutcomeType = na_if(maternalOutcomeType, \"\")) %>%\n",
    "  filter(!is.na(maternalOutcomeType))\n",
    "\n",
    "#trim white space\n",
    "datMat_long$maternalOutcomeType <-trimws(datMat_long$maternalOutcomeType)\n",
    "datMat_long <- datMat_long[,-2]\n",
    "\n",
    "#Remove all \"None\" entries (~NA in our data)\n",
    "datMat_long[datMat_long == \"None\"] <- NA\n",
    "datMat_long <- na.omit(datMat_long)\n",
    "\n",
    "#View data\n",
    "head(datMat_long[order(datMat_long$studyRef),])\n",
    "\n",
    "###### Offspring data\n",
    "datOff<-read.csv(\"tblStudyOffspringOutcomes.csv\",header=T)\n",
    "\n",
    "# long format with multiple nutrient types per study shown in multiple rows\n",
    "datOff_long <- datOff %>%\n",
    "  pivot_longer(cols = -studyRef, values_to = \"offspringOutcomeType\") %>%\n",
    "  mutate(offspringOutcomeType = na_if(offspringOutcomeType, \"\")) %>%\n",
    "  filter(!is.na(offspringOutcomeType))\n",
    "\n",
    "#trim white space\n",
    "datOff_long$offspringOutcomeType <-trimws(datOff_long$offspringOutcomeType)\n",
    "datOff_long <- datOff_long[,-2]\n",
    "\n",
    "#Remove all \"None\" entries (~NA in our data)\n",
    "datOff_long[datOff_long == \"None\"] <- NA\n",
    "datOff_long <- na.omit(datOff_long)\n",
    "\n",
    "\n",
    "#View data\n",
    "head(datOff_long[order(datOff_long$studyRef),])\n",
    "unique(datOff_long$offspringOutcomeType)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "73d44e34-b76d-40fa-ab37-9e3f3372f681",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Birth weight'</li><li>'Apgar score'</li><li>'Gestational age (Large or Small)'</li><li>'NICU admission'</li><li>'Other - maternal'</li><li>'Gestational age (Days/Weeks)'</li><li>'Cord blood pH'</li><li>'Respiratory distress'</li><li>'Congenital malformations'</li><li>'Congenital heart disease'</li><li>'Growth'</li><li>'Macrosomia'</li><li>'Foetal distress'</li><li>'Neonatal death'</li><li>'COVID status'</li><li>'Behavioural change'</li><li>'Neonate length'</li><li>'Head circumference'</li><li>'Cleft palate'</li><li>'Microcephaly'</li><li>'Gene expression - offspring'</li><li>'Language and motor development'</li><li>'Biomarkers of immunity - offspring'</li><li>'Autism spectrum disorder'</li><li>'Motor development'</li><li>'Neonatal encephalopathy'</li><li>'Mode of delivery / caesarean section rate'</li><li>'Pre-term birth'</li><li>'Maternal mortality'</li><li>'Stillbirth'</li><li>'Gene expression - maternal'</li><li>'Pre-eclampsia'</li><li>'Hypertension (pregnancy-induced)'</li><li>'Anaemia - maternal'</li><li>'Parasite load'</li><li>'Gestational diabetes'</li><li>'Pre-term pre-labour rupture of membranes (PPROM)'</li><li>'Haemorrhage'</li><li>'Miscarriage / spontaneous abortion'</li><li>'Perinatal mortality'</li><li>'Biomarkers of immunity - maternal'</li><li>'Placental inflammation'</li><li>'Hospitalisation or ICU admission'</li><li>'Haemoglobin levels - maternal'</li><li>'Postterm pregnancy'</li><li>'Intrauterine growth restriction (IUGR)'</li><li>'Placental abruption'</li><li>'Maternal mental health'</li><li>'Cancer risk'</li><li>'Venous thromboembolism / pulmonary embolism / deep vein thrombosis'</li><li>'PROM'</li><li>'Prolonged labour (labour dystocia)'</li><li>'Hyperemesis gravidarum'</li><li>'Placental size'</li><li>'DNA methylation - maternal'</li><li>'Placental lesions'</li><li>'Ectopic pregnancy'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Birth weight'\n",
       "\\item 'Apgar score'\n",
       "\\item 'Gestational age (Large or Small)'\n",
       "\\item 'NICU admission'\n",
       "\\item 'Other - maternal'\n",
       "\\item 'Gestational age (Days/Weeks)'\n",
       "\\item 'Cord blood pH'\n",
       "\\item 'Respiratory distress'\n",
       "\\item 'Congenital malformations'\n",
       "\\item 'Congenital heart disease'\n",
       "\\item 'Growth'\n",
       "\\item 'Macrosomia'\n",
       "\\item 'Foetal distress'\n",
       "\\item 'Neonatal death'\n",
       "\\item 'COVID status'\n",
       "\\item 'Behavioural change'\n",
       "\\item 'Neonate length'\n",
       "\\item 'Head circumference'\n",
       "\\item 'Cleft palate'\n",
       "\\item 'Microcephaly'\n",
       "\\item 'Gene expression - offspring'\n",
       "\\item 'Language and motor development'\n",
       "\\item 'Biomarkers of immunity - offspring'\n",
       "\\item 'Autism spectrum disorder'\n",
       "\\item 'Motor development'\n",
       "\\item 'Neonatal encephalopathy'\n",
       "\\item 'Mode of delivery / caesarean section rate'\n",
       "\\item 'Pre-term birth'\n",
       "\\item 'Maternal mortality'\n",
       "\\item 'Stillbirth'\n",
       "\\item 'Gene expression - maternal'\n",
       "\\item 'Pre-eclampsia'\n",
       "\\item 'Hypertension (pregnancy-induced)'\n",
       "\\item 'Anaemia - maternal'\n",
       "\\item 'Parasite load'\n",
       "\\item 'Gestational diabetes'\n",
       "\\item 'Pre-term pre-labour rupture of membranes (PPROM)'\n",
       "\\item 'Haemorrhage'\n",
       "\\item 'Miscarriage / spontaneous abortion'\n",
       "\\item 'Perinatal mortality'\n",
       "\\item 'Biomarkers of immunity - maternal'\n",
       "\\item 'Placental inflammation'\n",
       "\\item 'Hospitalisation or ICU admission'\n",
       "\\item 'Haemoglobin levels - maternal'\n",
       "\\item 'Postterm pregnancy'\n",
       "\\item 'Intrauterine growth restriction (IUGR)'\n",
       "\\item 'Placental abruption'\n",
       "\\item 'Maternal mental health'\n",
       "\\item 'Cancer risk'\n",
       "\\item 'Venous thromboembolism / pulmonary embolism / deep vein thrombosis'\n",
       "\\item 'PROM'\n",
       "\\item 'Prolonged labour (labour dystocia)'\n",
       "\\item 'Hyperemesis gravidarum'\n",
       "\\item 'Placental size'\n",
       "\\item 'DNA methylation - maternal'\n",
       "\\item 'Placental lesions'\n",
       "\\item 'Ectopic pregnancy'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Birth weight'\n",
       "2. 'Apgar score'\n",
       "3. 'Gestational age (Large or Small)'\n",
       "4. 'NICU admission'\n",
       "5. 'Other - maternal'\n",
       "6. 'Gestational age (Days/Weeks)'\n",
       "7. 'Cord blood pH'\n",
       "8. 'Respiratory distress'\n",
       "9. 'Congenital malformations'\n",
       "10. 'Congenital heart disease'\n",
       "11. 'Growth'\n",
       "12. 'Macrosomia'\n",
       "13. 'Foetal distress'\n",
       "14. 'Neonatal death'\n",
       "15. 'COVID status'\n",
       "16. 'Behavioural change'\n",
       "17. 'Neonate length'\n",
       "18. 'Head circumference'\n",
       "19. 'Cleft palate'\n",
       "20. 'Microcephaly'\n",
       "21. 'Gene expression - offspring'\n",
       "22. 'Language and motor development'\n",
       "23. 'Biomarkers of immunity - offspring'\n",
       "24. 'Autism spectrum disorder'\n",
       "25. 'Motor development'\n",
       "26. 'Neonatal encephalopathy'\n",
       "27. 'Mode of delivery / caesarean section rate'\n",
       "28. 'Pre-term birth'\n",
       "29. 'Maternal mortality'\n",
       "30. 'Stillbirth'\n",
       "31. 'Gene expression - maternal'\n",
       "32. 'Pre-eclampsia'\n",
       "33. 'Hypertension (pregnancy-induced)'\n",
       "34. 'Anaemia - maternal'\n",
       "35. 'Parasite load'\n",
       "36. 'Gestational diabetes'\n",
       "37. 'Pre-term pre-labour rupture of membranes (PPROM)'\n",
       "38. 'Haemorrhage'\n",
       "39. 'Miscarriage / spontaneous abortion'\n",
       "40. 'Perinatal mortality'\n",
       "41. 'Biomarkers of immunity - maternal'\n",
       "42. 'Placental inflammation'\n",
       "43. 'Hospitalisation or ICU admission'\n",
       "44. 'Haemoglobin levels - maternal'\n",
       "45. 'Postterm pregnancy'\n",
       "46. 'Intrauterine growth restriction (IUGR)'\n",
       "47. 'Placental abruption'\n",
       "48. 'Maternal mental health'\n",
       "49. 'Cancer risk'\n",
       "50. 'Venous thromboembolism / pulmonary embolism / deep vein thrombosis'\n",
       "51. 'PROM'\n",
       "52. 'Prolonged labour (labour dystocia)'\n",
       "53. 'Hyperemesis gravidarum'\n",
       "54. 'Placental size'\n",
       "55. 'DNA methylation - maternal'\n",
       "56. 'Placental lesions'\n",
       "57. 'Ectopic pregnancy'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Birth weight\"                                                      \n",
       " [2] \"Apgar score\"                                                       \n",
       " [3] \"Gestational age (Large or Small)\"                                  \n",
       " [4] \"NICU admission\"                                                    \n",
       " [5] \"Other - maternal\"                                                  \n",
       " [6] \"Gestational age (Days/Weeks)\"                                      \n",
       " [7] \"Cord blood pH\"                                                     \n",
       " [8] \"Respiratory distress\"                                              \n",
       " [9] \"Congenital malformations\"                                          \n",
       "[10] \"Congenital heart disease\"                                          \n",
       "[11] \"Growth\"                                                            \n",
       "[12] \"Macrosomia\"                                                        \n",
       "[13] \"Foetal distress\"                                                   \n",
       "[14] \"Neonatal death\"                                                    \n",
       "[15] \"COVID status\"                                                      \n",
       "[16] \"Behavioural change\"                                                \n",
       "[17] \"Neonate length\"                                                    \n",
       "[18] \"Head circumference\"                                                \n",
       "[19] \"Cleft palate\"                                                      \n",
       "[20] \"Microcephaly\"                                                      \n",
       "[21] \"Gene expression - offspring\"                                       \n",
       "[22] \"Language and motor development\"                                    \n",
       "[23] \"Biomarkers of immunity - offspring\"                                \n",
       "[24] \"Autism spectrum disorder\"                                          \n",
       "[25] \"Motor development\"                                                 \n",
       "[26] \"Neonatal encephalopathy\"                                           \n",
       "[27] \"Mode of delivery / caesarean section rate\"                         \n",
       "[28] \"Pre-term birth\"                                                    \n",
       "[29] \"Maternal mortality\"                                                \n",
       "[30] \"Stillbirth\"                                                        \n",
       "[31] \"Gene expression - maternal\"                                        \n",
       "[32] \"Pre-eclampsia\"                                                     \n",
       "[33] \"Hypertension (pregnancy-induced)\"                                  \n",
       "[34] \"Anaemia - maternal\"                                                \n",
       "[35] \"Parasite load\"                                                     \n",
       "[36] \"Gestational diabetes\"                                              \n",
       "[37] \"Pre-term pre-labour rupture of membranes (PPROM)\"                  \n",
       "[38] \"Haemorrhage\"                                                       \n",
       "[39] \"Miscarriage / spontaneous abortion\"                                \n",
       "[40] \"Perinatal mortality\"                                               \n",
       "[41] \"Biomarkers of immunity - maternal\"                                 \n",
       "[42] \"Placental inflammation\"                                            \n",
       "[43] \"Hospitalisation or ICU admission\"                                  \n",
       "[44] \"Haemoglobin levels - maternal\"                                     \n",
       "[45] \"Postterm pregnancy\"                                                \n",
       "[46] \"Intrauterine growth restriction (IUGR)\"                            \n",
       "[47] \"Placental abruption\"                                               \n",
       "[48] \"Maternal mental health\"                                            \n",
       "[49] \"Cancer risk\"                                                       \n",
       "[50] \"Venous thromboembolism / pulmonary embolism / deep vein thrombosis\"\n",
       "[51] \"PROM\"                                                              \n",
       "[52] \"Prolonged labour (labour dystocia)\"                                \n",
       "[53] \"Hyperemesis gravidarum\"                                            \n",
       "[54] \"Placental size\"                                                    \n",
       "[55] \"DNA methylation - maternal\"                                        \n",
       "[56] \"Placental lesions\"                                                 \n",
       "[57] \"Ectopic pregnancy\"                                                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Other - maternal'</li><li>'Anaemia - maternal'</li><li>'Parasite load'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Other - maternal'\n",
       "\\item 'Anaemia - maternal'\n",
       "\\item 'Parasite load'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Other - maternal'\n",
       "2. 'Anaemia - maternal'\n",
       "3. 'Parasite load'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"Other - maternal\"   \"Anaemia - maternal\" \"Parasite load\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Anaemia'</li><li>'Haemoglobin levels - offspring'</li><li>'DNA methylation - maternal - offspring'</li><li>'Other'</li><li>'None'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Anaemia'\n",
       "\\item 'Haemoglobin levels - offspring'\n",
       "\\item 'DNA methylation - maternal - offspring'\n",
       "\\item 'Other'\n",
       "\\item 'None'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Anaemia'\n",
       "2. 'Haemoglobin levels - offspring'\n",
       "3. 'DNA methylation - maternal - offspring'\n",
       "4. 'Other'\n",
       "5. 'None'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"Anaemia\"                               \n",
       "[2] \"Haemoglobin levels - offspring\"        \n",
       "[3] \"DNA methylation - maternal - offspring\"\n",
       "[4] \"Other\"                                 \n",
       "[5] \"None\"                                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>pregnancyOutcomeType</th><th scope=col>studyRef</th><th scope=col>Category</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>125</th><td>Behavioural change               </td><td>1182475095</td><td>Developmental and Long-term Child Outcomes</td></tr>\n",
       "\t<tr><th scope=row>287</th><td>Biomarkers of immunity - maternal</td><td>1182475095</td><td>Placental and Immunological Factors       </td></tr>\n",
       "\t<tr><th scope=row>1841</th><td>Other - maternal                 </td><td>1182475123</td><td>NA                                        </td></tr>\n",
       "\t<tr><th scope=row>1967</th><td>Placental inflammation           </td><td>1182475123</td><td>Placental and Immunological Factors       </td></tr>\n",
       "\t<tr><th scope=row>14</th><td>Anaemia - maternal               </td><td>1182475134</td><td>NA                                        </td></tr>\n",
       "\t<tr><th scope=row>12</th><td>Anaemia - maternal               </td><td>1182475144</td><td>NA                                        </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & pregnancyOutcomeType & studyRef & Category\\\\\n",
       "  & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t125 & Behavioural change                & 1182475095 & Developmental and Long-term Child Outcomes\\\\\n",
       "\t287 & Biomarkers of immunity - maternal & 1182475095 & Placental and Immunological Factors       \\\\\n",
       "\t1841 & Other - maternal                  & 1182475123 & NA                                        \\\\\n",
       "\t1967 & Placental inflammation            & 1182475123 & Placental and Immunological Factors       \\\\\n",
       "\t14 & Anaemia - maternal                & 1182475134 & NA                                        \\\\\n",
       "\t12 & Anaemia - maternal                & 1182475144 & NA                                        \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | pregnancyOutcomeType &lt;chr&gt; | studyRef &lt;int&gt; | Category &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 125 | Behavioural change                | 1182475095 | Developmental and Long-term Child Outcomes |\n",
       "| 287 | Biomarkers of immunity - maternal | 1182475095 | Placental and Immunological Factors        |\n",
       "| 1841 | Other - maternal                  | 1182475123 | NA                                         |\n",
       "| 1967 | Placental inflammation            | 1182475123 | Placental and Immunological Factors        |\n",
       "| 14 | Anaemia - maternal                | 1182475134 | NA                                         |\n",
       "| 12 | Anaemia - maternal                | 1182475144 | NA                                         |\n",
       "\n"
      ],
      "text/plain": [
       "     pregnancyOutcomeType              studyRef  \n",
       "125  Behavioural change                1182475095\n",
       "287  Biomarkers of immunity - maternal 1182475095\n",
       "1841 Other - maternal                  1182475123\n",
       "1967 Placental inflammation            1182475123\n",
       "14   Anaemia - maternal                1182475134\n",
       "12   Anaemia - maternal                1182475144\n",
       "     Category                                  \n",
       "125  Developmental and Long-term Child Outcomes\n",
       "287  Placental and Immunological Factors       \n",
       "1841 NA                                        \n",
       "1967 Placental and Immunological Factors       \n",
       "14   NA                                        \n",
       "12   NA                                        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Combine the two datasets into a list of Pregnancy Outcomes. \n",
    "# use the PregnancyOutcomes lookup table to create a Pregnancy Outcomes variable df. \n",
    "\n",
    "#1. Rename columns to facilitate rbind\n",
    "names(datOff_long)[2] <- \"pregnancyOutcomeType\"\n",
    "names(datMat_long)[2] <- \"pregnancyOutcomeType\"\n",
    "\n",
    "datPreg_long <- bind_rows(datOff_long, datMat_long)\n",
    "\n",
    "\n",
    "\n",
    "#What are all the pregnancyOutcomeTypes?\n",
    "unique(datPreg_long$pregnancyOutcomeType)\n",
    "\n",
    "# add categorisation\n",
    "# Lookup table with shortened category names\n",
    "\n",
    "##Notes: \n",
    "#### None means no data/NA\n",
    "\n",
    "\n",
    "pregnancy_lookup <- data.frame(Pregnancy = c(\"Mode of delivery / caesarean section rate\", \"Maternal mortality\", \"Hypertension (pregnancy-induced)\",\n",
    "\"Pre-eclampsia\", \"Anaemia\", \"Gestational diabetes\", \"Haemorrhage\", \"Hospitalisation or ICU admission\", \"Postterm pregnancy\",\n",
    "\"Placental abruption\", \"Maternal mental health\", \"Cancer risk\", \"Venous thromboembolism / pulmonary embolism / deep vein thrombosis\",\n",
    "\"Prolonged labour (labour dystocia)\", \"Hyperemesis gravidarum\", \"Ectopic pregnancy\", \"COVID status\", \n",
    "\"Stillbirth\", \"Miscarriage / spontaneous abortion\",\n",
    "\"Perinatal mortality\", \"Neonatal death\", \"NICU admission\", \"Birth weight\", \"Gestational age (Large or Small)\", \"Gestational age (Days/Weeks)\",\n",
    "\"Apgar score\", \"Cord blood pH\", \"Respiratory distress\", \"Congenital malformations\", \"Congenital heart disease\", \"Foetal distress\", \"Macrosomia\",\n",
    "\"Intrauterine growth restriction (IUGR)\", \"Neonatal encephalopathy\", \"Cleft palate\", \"Microcephaly\", \"Neonate length\", \"Head circumference\",\n",
    "\"Pre-term birth\", \"Pre-term pre-labour rupture of membranes (PPROM)\", \"PROM\", \"Placental inflammation\", \"Placental size\", \"Placental lesions\",\n",
    "\"Biomarkers of immunity - maternal\", \"Biomarkers of immunity - offspring\", \"Haemoglobin levels - maternal\", \"Haemoglobin levels - offspring\", \n",
    "\"Gene expression - maternal\", \"Gene expression - offspring\", \"DNA methylation - maternal\", \"DNA methylation - maternal - offspring\", \n",
    "\"Growth\", \"Language and motor development\", \"Motor development\",\n",
    "\"Autism spectrum disorder\", \"Behavioural change\", \"Other\", \"None\"),\n",
    "Category = c(\n",
    "rep(\"Maternal Outcomes\", 17),\n",
    "rep(\"Foetal and Neonatal Outcomes\", 21),\n",
    "rep(\"Pregnancy Complications\", 3),\n",
    "rep(\"Placental and Immunological Factors\", 5),\n",
    "rep(\"Molecular and Clinical Biomarkers\", 6),\n",
    "rep(\"Developmental and Long-term Child Outcomes\", 5),\n",
    "rep(\"Other\", 1),\n",
    "rep(\"None\", 1)\n",
    "),\n",
    "    stringsAsFactors = FALSE\n",
    ")\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#Check that all the nutrition types have been categorised\n",
    "setdiff(unique(datPreg_long$pregnancyOutcomeType), unique(pregnancy_lookup$Pregnancy))\n",
    "setdiff(unique(pregnancy_lookup$Pregnancy), unique(datPreg_long$pregnancyOutcomeType))\n",
    "\n",
    "#Create catagorised dataset\n",
    "# Left join nutrient_lookup with datNut\n",
    "datPregType_long <- merge(datPreg_long, pregnancy_lookup, by.x = \"pregnancyOutcomeType\", by.y = \"Pregnancy\", all.x = TRUE)\n",
    "\n",
    "head(datPregType_long[order(datPregType_long$studyRef),])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e4d6331d-6b4a-4f25-aca8-b86be3e3c4e5",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "# Full join pregnancy Outcome with Meta, Nutrition, and Infection\n",
    "\n",
    "This creates one big table with all the input and output variables for a specific study. \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "3c063dcf-3352-4cc5-9e42-28bb7426bb59",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "\u001b[1m\u001b[33mError\u001b[39m in `rename()`:\u001b[22m\n\u001b[33m!\u001b[39m Can't rename columns that don't exist.\n\u001b[31m✖\u001b[39m Column `NutCat` doesn't exist.\n",
     "output_type": "error",
     "traceback": [
      "\u001b[1m\u001b[33mError\u001b[39m in `rename()`:\u001b[22m\n\u001b[33m!\u001b[39m Can't rename columns that don't exist.\n\u001b[31m✖\u001b[39m Column `NutCat` doesn't exist.\nTraceback:\n",
      "1. rename(datMetaNutInfPreg, Category.x = \"NutCat\", Category.y = \"PregCat\")",
      "2. rename.data.frame(datMetaNutInfPreg, Category.x = \"NutCat\", Category.y = \"PregCat\")",
      "3. tidyselect::eval_rename(expr(c(...)), .data)",
      "4. rename_impl(data, names(data), as_quosure(expr, env), strict = strict, \n .     name_spec = name_spec, allow_predicates = allow_predicates, \n .     error_call = error_call)",
      "5. eval_select_impl(x, names, {\n .     {\n .         sel\n .     }\n . }, strict = strict, name_spec = name_spec, type = \"rename\", allow_predicates = allow_predicates, \n .     error_call = error_call)",
      "6. with_subscript_errors(out <- vars_select_eval(vars, expr, strict = strict, \n .     data = x, name_spec = name_spec, uniquely_named = uniquely_named, \n .     allow_rename = allow_rename, allow_empty = allow_empty, allow_predicates = allow_predicates, \n .     type = type, error_call = error_call), type = type)",
      "7. try_fetch(expr, vctrs_error_subscript = function(cnd) {\n .     cnd$subscript_action <- subscript_action(type)\n .     cnd$subscript_elt <- \"column\"\n .     cnd_signal(cnd)\n . })",
      "8. withCallingHandlers(expr, vctrs_error_subscript = function(cnd) {\n .     {\n .         .__handler_frame__. <- TRUE\n .         .__setup_frame__. <- frame\n .     }\n .     out <- handlers[[1L]](cnd)\n .     if (!inherits(out, \"rlang_zap\")) \n .         throw(out)\n . })",
      "9. vars_select_eval(vars, expr, strict = strict, data = x, name_spec = name_spec, \n .     uniquely_named = uniquely_named, allow_rename = allow_rename, \n .     allow_empty = allow_empty, allow_predicates = allow_predicates, \n .     type = type, error_call = error_call)",
      "10. walk_data_tree(expr, data_mask, context_mask)",
      "11. eval_c(expr, data_mask, context_mask)",
      "12. reduce_sels(node, data_mask, context_mask, init = init)",
      "13. walk_data_tree(new, data_mask, context_mask)",
      "14. as_indices_sel_impl(out, vars = vars, strict = strict, data = data, \n  .     allow_predicates = allow_predicates, call = error_call, arg = as_label(expr))",
      "15. as_indices_impl(x, vars, call = call, arg = arg, strict = strict)",
      "16. chr_as_locations(x, vars, call = call, arg = arg)",
      "17. vctrs::vec_as_location(x, n = length(vars), names = vars, call = call, \n  .     arg = arg)",
      "18. (function () \n  . stop_subscript_oob(i = i, subscript_type = subscript_type, names = names, \n  .     subscript_action = subscript_action, subscript_arg = subscript_arg, \n  .     call = call))()",
      "19. stop_subscript_oob(i = i, subscript_type = subscript_type, names = names, \n  .     subscript_action = subscript_action, subscript_arg = subscript_arg, \n  .     call = call)",
      "20. stop_subscript(class = \"vctrs_error_subscript_oob\", i = i, subscript_type = subscript_type, \n  .     ..., call = call)",
      "21. abort(class = c(class, \"vctrs_error_subscript\"), i = i, ..., \n  .     call = vctrs_error_call(call))",
      "22. signal_abort(cnd, .file)",
      "23. signalCondition(cnd)",
      "24. (function (cnd) \n  . {\n  .     {\n  .         .__handler_frame__. <- TRUE\n  .         .__setup_frame__. <- frame\n  .     }\n  .     out <- handlers[[1L]](cnd)\n  .     if (!inherits(out, \"rlang_zap\")) \n  .         throw(out)\n  . })(structure(list(message = \"\", trace = structure(list(call = list(\n  .     IRkernel::main(), kernel$run(), handle_shell(), executor$execute(msg), \n  .     tryCatch(evaluate(request$content$code, envir = .GlobalEnv, \n  .         output_handler = oh, stop_on_error = 1L), interrupt = function(cond) {\n  .         log_debug(\"Interrupt during execution\")\n  .         interrupted <<- TRUE\n  .     }, error = .self$handle_error), tryCatchList(expr, classes, \n  .         parentenv, handlers), tryCatchOne(tryCatchList(expr, \n  .         names[-nh], parentenv, handlers[-nh]), names[nh], parentenv, \n  .         handlers[[nh]]), doTryCatch(return(expr), name, parentenv, \n  .         handler), tryCatchList(expr, names[-nh], parentenv, handlers[-nh]), \n  .     tryCatchOne(expr, names, parentenv, handlers[[1L]]), doTryCatch(return(expr), \n  .         name, parentenv, handler), evaluate(request$content$code, \n  .         envir = .GlobalEnv, output_handler = oh, stop_on_error = 1L), \n  .     evaluate_call(expr, parsed$src[[i]], envir = envir, enclos = enclos, \n  .         debug = debug, last = i == length(out), use_try = stop_on_error != \n  .             2L, keep_warning = keep_warning, keep_message = keep_message, \n  .         output_handler = output_handler, include_timing = include_timing), \n  .     timing_fn(handle(ev <- withCallingHandlers(withVisible(eval_with_user_handlers(expr, \n  .         envir, enclos, user_handlers)), warning = wHandler, error = eHandler, \n  .         message = mHandler))), handle(ev <- withCallingHandlers(withVisible(eval_with_user_handlers(expr, \n  .         envir, enclos, user_handlers)), warning = wHandler, error = eHandler, \n  .         message = mHandler)), try(f, silent = TRUE), tryCatch(expr, \n  .         error = function(e) {\n  .             call <- conditionCall(e)\n  .             if (!is.null(call)) {\n  .                 if (identical(call[[1L]], quote(doTryCatch))) \n  .                   call <- sys.call(-4L)\n  .                 dcall <- deparse(call, nlines = 1L)\n  .                 prefix <- paste(\"Error in\", dcall, \": \")\n  .                 LONG <- 75L\n  .                 sm <- strsplit(conditionMessage(e), \"\\n\")[[1L]]\n  .                 w <- 14L + nchar(dcall, type = \"w\") + nchar(sm[1L], \n  .                   type = \"w\")\n  .                 if (is.na(w)) \n  .                   w <- 14L + nchar(dcall, type = \"b\") + nchar(sm[1L], \n  .                     type = \"b\")\n  .                 if (w > LONG) \n  .                   prefix <- paste0(prefix, \"\\n  \")\n  .             }\n  .             else prefix <- \"Error : \"\n  .             msg <- paste0(prefix, conditionMessage(e), \"\\n\")\n  .             .Internal(seterrmessage(msg[1L]))\n  .             if (!silent && isTRUE(getOption(\"show.error.messages\"))) {\n  .                 cat(msg, file = outFile)\n  .                 .Internal(printDeferredWarnings())\n  .             }\n  .             invisible(structure(msg, class = \"try-error\", condition = e))\n  .         }), tryCatchList(expr, classes, parentenv, handlers), \n  .     tryCatchOne(expr, names, parentenv, handlers[[1L]]), doTryCatch(return(expr), \n  .         name, parentenv, handler), withCallingHandlers(withVisible(eval_with_user_handlers(expr, \n  .         envir, enclos, user_handlers)), warning = wHandler, error = eHandler, \n  .         message = mHandler), withVisible(eval_with_user_handlers(expr, \n  .         envir, enclos, user_handlers)), eval_with_user_handlers(expr, \n  .         envir, enclos, user_handlers), eval(expr, envir, enclos), \n  .     eval(expr, envir, enclos), rename(datMetaNutInfPreg, Category.x = \"NutCat\", \n  .         Category.y = \"PregCat\"), rename.data.frame(datMetaNutInfPreg, \n  .         Category.x = \"NutCat\", Category.y = \"PregCat\"), tidyselect::eval_rename(expr(c(...)), \n  .         .data), rename_impl(data, names(data), as_quosure(expr, \n  .         env), strict = strict, name_spec = name_spec, allow_predicates = allow_predicates, \n  .         error_call = error_call), eval_select_impl(x, names, \n  .         {\n  .             {\n  .                 sel\n  .             }\n  .         }, strict = strict, name_spec = name_spec, type = \"rename\", \n  .         allow_predicates = allow_predicates, error_call = error_call), \n  .     with_subscript_errors(out <- vars_select_eval(vars, expr, \n  .         strict = strict, data = x, name_spec = name_spec, uniquely_named = uniquely_named, \n  .         allow_rename = allow_rename, allow_empty = allow_empty, \n  .         allow_predicates = allow_predicates, type = type, error_call = error_call), \n  .         type = type), try_fetch(expr, vctrs_error_subscript = function(cnd) {\n  .         cnd$subscript_action <- subscript_action(type)\n  .         cnd$subscript_elt <- \"column\"\n  .         cnd_signal(cnd)\n  .     }), withCallingHandlers(expr, vctrs_error_subscript = function(cnd) {\n  .         {\n  .             .__handler_frame__. <- TRUE\n  .             .__setup_frame__. <- frame\n  .         }\n  .         out <- handlers[[1L]](cnd)\n  .         if (!inherits(out, \"rlang_zap\")) \n  .             throw(out)\n  .     }), vars_select_eval(vars, expr, strict = strict, data = x, \n  .         name_spec = name_spec, uniquely_named = uniquely_named, \n  .         allow_rename = allow_rename, allow_empty = allow_empty, \n  .         allow_predicates = allow_predicates, type = type, error_call = error_call), \n  .     walk_data_tree(expr, data_mask, context_mask), eval_c(expr, \n  .         data_mask, context_mask), reduce_sels(node, data_mask, \n  .         context_mask, init = init), walk_data_tree(new, data_mask, \n  .         context_mask), as_indices_sel_impl(out, vars = vars, \n  .         strict = strict, data = data, allow_predicates = allow_predicates, \n  .         call = error_call, arg = as_label(expr)), as_indices_impl(x, \n  .         vars, call = call, arg = arg, strict = strict), chr_as_locations(x, \n  .         vars, call = call, arg = arg), vctrs::vec_as_location(x, \n  .         n = length(vars), names = vars, call = call, arg = arg), \n  .     `<fn>`(), stop_subscript_oob(i = i, subscript_type = subscript_type, \n  .         names = names, subscript_action = subscript_action, subscript_arg = subscript_arg, \n  .         call = call), stop_subscript(class = \"vctrs_error_subscript_oob\", \n  .         i = i, subscript_type = subscript_type, ..., call = call), \n  .     abort(class = c(class, \"vctrs_error_subscript\"), i = i, ..., \n  .         call = vctrs_error_call(call))), parent = c(0L, 1L, 2L, \n  . 3L, 4L, 5L, 6L, 7L, 6L, 9L, 10L, 4L, 12L, 13L, 13L, 15L, 16L, \n  . 17L, 18L, 19L, 13L, 13L, 13L, 23L, 24L, 0L, 0L, 27L, 28L, 29L, \n  . 30L, 31L, 32L, 30L, 34L, 35L, 36L, 37L, 38L, 39L, 40L, 41L, 0L, \n  . 43L, 44L, 45L), visible = c(TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, \n  . TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, \n  . TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, FALSE, \n  . FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, \n  . FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE\n  . ), namespace = c(\"IRkernel\", NA, \"IRkernel\", NA, \"base\", \"base\", \n  . \"base\", \"base\", \"base\", \"base\", \"base\", \"evaluate\", \"evaluate\", \n  . \"evaluate\", \"evaluate\", \"base\", \"base\", \"base\", \"base\", \"base\", \n  . \"base\", \"base\", \"evaluate\", \"base\", \"base\", \"dplyr\", \"dplyr\", \n  . \"tidyselect\", \"tidyselect\", \"tidyselect\", \"tidyselect\", \"rlang\", \n  . \"base\", \"tidyselect\", \"tidyselect\", \"tidyselect\", \"tidyselect\", \n  . \"tidyselect\", \"tidyselect\", \"tidyselect\", \"tidyselect\", \"vctrs\", \n  . \"vctrs\", \"vctrs\", \"vctrs\", \"rlang\"), scope = c(\"::\", NA, \"local\", \n  . NA, \"::\", \"local\", \"local\", \"local\", \"local\", \"local\", \"local\", \n  . \"::\", \":::\", \"local\", \"local\", \"::\", \"::\", \"local\", \"local\", \n  . \"local\", \"::\", \"::\", \":::\", \"::\", \"::\", \"::\", \":::\", \"::\", \":::\", \n  . \":::\", \":::\", \"::\", \"::\", \":::\", \":::\", \":::\", \":::\", \":::\", \n  . \":::\", \":::\", \":::\", \"::\", \"local\", \":::\", \":::\", \"::\"), error_frame = c(FALSE, \n  . FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, \n  . FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, \n  . FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, TRUE, FALSE, \n  . FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, \n  . FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE\n  . )), row.names = c(NA, -46L), version = 2L, class = c(\"rlang_trace\", \n  . \"rlib_trace\", \"tbl\", \"data.frame\")), parent = NULL, i = \"NutCat\", \n  .     subscript_type = \"character\", names = c(\"studyRef\", \"assessedBy\", \n  .     \"firstAuthor\", \"year\", \"species\", \"countryStudy\", \"mainOutcome\", \n  .     \"studyScore\", \"nutritionType\", \"Category.x\", \"infectionType\", \n  .     \"InfCat\", \"pregnancyOutcomeType\", \"Category.y\"), subscript_action = NULL, \n  .     subscript_arg = \"\\\"NutCat\\\"\", call = rename(datMetaNutInfPreg, \n  .         Category.x = \"NutCat\", Category.y = \"PregCat\")), class = c(\"vctrs_error_subscript_oob\", \n  . \"vctrs_error_subscript\", \"rlang_error\", \"error\", \"condition\")))",
      "25. handlers[[1L]](cnd)",
      "26. cnd_signal(cnd)",
      "27. signal_abort(cnd)"
     ]
    }
   ],
   "source": [
    "#List of all the dfs\n",
    "\n",
    "dfs_list <- list(datMeta, datNutType_long, datInfType_long, datPregType_long)\n",
    "\n",
    "# Full join all data frames by \"studyRef\"\n",
    "\n",
    "datMetaNutInfPreg <- Reduce(function(x,y) full_join(x,y, by = \"studyRef\"), dfs_list)\n",
    "datMetaNutInfPreg <- rename(datMetaNutInfPreg, \"NutCat\", \"Category.y\" = \"PregCat\")\n",
    "\n",
    "                            \n",
    "head(datMetaNutInfPreg)\n",
    "dim(datMetaNutInfPreg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46378ea5-bf79-41a2-8175-420daf284c59",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "abdf6c0a-b737-48e5-a6b3-24528fd92c6d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
